gobblin-metrics/src/main/java/gobblin/metrics/reporter/ScheduledReporter.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.metrics.reporter;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.Timer;
+import com.google.common.base.Optional;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigValueFactory;
+
+import gobblin.metrics.InnerMetricContext;
+import gobblin.metrics.context.ReportableContext;
+import gobblin.util.ExecutorsUtils;
+
+
+/**
+ * A {@link ContextAwareReporter} that reports on a schedule.
+ */
+@Slf4j
+public abstract class ScheduledReporter extends ContextAwareReporter {
+
+  public static final String REPORTING_INTERVAL_PERIOD = "reporting.interval.period";
+  public static final long DEFAULT_REPORTING_INTERVAL_PERIOD = 1;
+  public static final String REPORTING_INTERVAL_UNIT = "reporting.interval.unit";
```

Comment:
How about we use the jodatime Period for parsing a string representation of period into reporting.interval.period ? Thus we can avoid the reporting.interval.unit config.